### PR TITLE
feat: Get Current Analytics Tracking Identifier (WPB-10600)

### PR DESCRIPTION
**Pull request by alexandreferris**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10600" title="WPB-10600" target="_blank">TaskWPB-10600</a>  [Android] Show countly details in debug settings
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

There was no possible way of getting current analytics tracking identifier directly (only existing way is a observer/flow)

### Causes (Optional)

Not implemented

### Solutions

Implement usecase and add tests

### Dependencies (Optional)

Needs releases with:

AR PR will follow

#### Test Coverage (Optional)

- [X] I have added automated test to this contribution

#### How to Test

There is no way to test this PR for now, as it will be used in conjunction with AR

### CODE REVIEW

**Review by yamilmedina — APPROVED** [review]

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/analytics/GetCurrentAnalyticsTrackingIdentifierUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * Use case [GetCurrentAnalyticsTrackingIdentifierUseCase] operation
+     *
+     * @return a [String] containing the current tracking identifier
+     */
+    suspend operator fun invoke(): String?
+}
+
+@Suppress("FunctionNaming")
+internal fun GetCurrentAnalyticsTrackingIdentifierUseCase(
```
Comment (borichellow) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/analytics/GetCurrentAnalyticsTrackingIdentifierUseCase.kt:35]: why not use common `internal class ...UseCaseImpl` pattern?

**Review by borichellow — APPROVED** [review]
just a small question :)

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/analytics/GetCurrentAnalyticsTrackingIdentifierUseCase.kt (hunk 1 above)
Comment (alexandreferris) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/analytics/GetCurrentAnalyticsTrackingIdentifierUseCase.kt:35]: So, Vitor started with this so we can remove the suffix of "Impl" (along with not having a prefix of "I" when having interfaces)

Comment (borichellow) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/analytics/GetCurrentAnalyticsTrackingIdentifierUseCase.kt:35]: hmmm, make sense :)

**Pull request merged**